easybuild/easyblocks/generic/intelbase.py

```diff
@@ -131,10 +131,9 @@ def parse_components_list(self):
                                      % self.cfg['components'])
         else:
             self.install_components = []
-            for comp in available_components:
-                for comp_regex in self.cfg['components']:
-                    if re.search(comp_regex, comp):
-                        self.install_components.append(comp)
+            for comp_regex in self.cfg['components']:
+                comps = [comp for comp in available_components if re.match(comp_regex, comp)]
```

Comment:
No, `comp` as regex will work for anything with `comp` in it: `comp-intel`, `intel-comp`